src/crawlers/browser_crawler.js

```diff
@@ -367,7 +367,7 @@ export default class BrowserCrawler extends BasicCrawler {
         };
 
         if (this.proxyConfiguration && this.launchContext && this.launchContext.useIncognitoPages) {
-            const proxyInfo = this.proxyConfiguration.newProxyInfo(session.id);
+            const proxyInfo = this.proxyConfiguration.newProxyInfo(session && session.id);
```

Comment:
we got engine `>=15` in package.json, it's safe to use optional chaining I guess, for both lines

Reply:
I agree. However let's keep this for consistency and wait for v3.